### ISSUE TITLE
“MicrosoftCognitiveServicesSpeech.xcframework” gives link error on integrating in iOS SDK framework

### ISSUE THREAD

**Issue by sagarmahewar** [Azure-Samples/cognitive-services-speech-sdk#1795]
**Describe the bug**
While using “MicrosoftCognitiveServicesSpeech.xcframework” in our IOS SDK framework, this gives us error as follows

——————————————————
ld: warning: directory not found for option '-L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.1.sdk/usr/lib/gcc/arm-apple-darwin10/4.0.1'
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_SPXAudioConfiguration", referenced from:
      objc-class-ref in XYZMedicalKit(NMXYZ.o)
  "_OBJC_CLASS_$_SPXCancellationDetails", referenced from:
      objc-class-ref in XYZMedicalKit(NMXYZ.o)
  "_OBJC_CLASS_$_SPXKeywordRecognitionModel", referenced from:
      objc-class-ref in XYZMedicalKit(NMXYZ.o)
  "_OBJC_CLASS_$_SPXKeywordRecognizer", referenced from:
      objc-class-ref in XYZMedicalKit(NMXYZ.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
——————————————————

Note - This works fine if we integrate it in “iOS Test Application”, only creates issue on integrating in “SDK framework”. 

**To Reproduce**
1. In iOS SDK
2. Add “MicrosoftCognitiveServicesSpeech.xcframework”
3. This gives link errors as mentioned above

**Expected behavior**
It should not give any link error

**Version of the Cognitive Services Speech SDK**
Latest

**Platform, Operating System, and Programming Language**
 - OS: iOS
 - Hardware - ARM64
 - Programming language: Objective-C
 - Browser : NA

**Additional context**
- Added details in description
- Is it Umbrella framework issue ?
- This is framework, does Microsoft also provide static library for same ? OR any other solution ?

**Comment by jhakulin** [comment]
@sagarmahewar Thanks for the report. Could you please tell us what is the version of Speech SDK you are using? 
Also please provide information did you got the build issue when building some of our Objective-C samples in this GitHub repository?

I just tested with 1.24.0 version of Speech SDK using the following quickstart for Objective-C that builds correctly without missing symbols.
https://github.com/Azure-Samples/cognitive-services-speech-sdk/tree/master/quickstart/objectivec/ios/from-microphone

Could you please let us know do you see a problem also with that quickstart?

The README.md file instruction could be improved there as it does not clearly inform customer where to place the xcframework package. The folder structure of the quickstart should be following
 
![quickstart](https://user-images.githubusercontent.com/16473503/210763842-62d6ec8c-51ee-49a1-b4ca-246444bd5eee.jpg)

**Comment by sagarmahewar** [comment]
Please find required info -

#1 Version is same as you tried “1.24.0”
#2 Haven’t faced any issue while building Objective-C samples from GitHub repo.
Note : Also we could integrate it in our test application without any issue. Only facing issues on integrating xcframework in another xcframework.
#3 No issue with QuickStart

**Comment by jhakulin** [comment]
@sagarmahewar Ok, now I see your problem better, cannot see the reason for the problem though. Would it be possible for you to share repro steps for the issue?

**Comment by sagarmahewar** [comment]
@jhakulin Please find steps as

#1 Create a XYZ.xcframework
#2A Add it in any of “Test/Sample” App and use it, Application should allow to access the content of XYZ.xcframework. OR
#2B Add MicrosoftCognitiveServicesSpeech.xcframework in “Test/Sample” App, we can access the MicrosoftCognitiveServicesSpeech features

#3 But IMPORTANT one, add “MicrosoftCognitiveServicesSpeech.xcframework” in that “XYZ.xcframework”
#4 Inside “XYZ.xcframework” write code to access the features of “MicrosoftCognitiveServicesSpeech”
#5A Now use same “Test/Sample” App with XYZ.xcframework
#5B Add some code in “Test/Sample” App, so that “XYZ.xcframework” internally could fetches the services of “MicrosoftCognitiveServicesSpeech.xcframework”

But on compiling “Test/Sample” App, it gives above mentioned thread errors.

In short 
#1 I can directly use “MicrosoftCognitiveServicesSpeech.xcframework” in “Test/Sample” App it works fine
#2A But when we add “MicrosoftCognitiveServicesSpeech.xcframework” in another xcframework I.e. “XYZ.xcframework” 
#2B and then, if the “Test/Sample” App uses “XYZ.xcframework” to access feature of MicrosoftCognitiveServicesSpeech, it gives error on build.

Is it possible to connect on call ? It will be helpful and quick. Thank you.

**Comment by jhakulin** [comment]
@sagarmahewar If you use your XYZ.xcframework and MicrosoftCognitiveServicesSpeech.xcframework separately in your app, does it work without linker errors?

**Comment by sagarmahewar** [comment]
@jhakulin Yes

**Comment by jhakulin** [comment]
@sagarmahewar As agreed in our previous discussion, please provide sample project which reproduces the issue.

**Comment by jhakulin** [comment]
Closing the issue as we have separate discussions in email to solve the issue
